### PR TITLE
Have CMake generate pkg-config file for library

### PR DESCRIPTION
**Pull request by JPeterMugaas**
So that various build processes can query for compiler and linker flags in Linux, MSYS2, and CygWin.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -237,6 +237,32 @@ install(FILES
     ${CMAKE_CURRENT_BINARY_DIR}/${PACKAGE_NAME}-config-version.cmake
   DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/${PACKAGE_NAME})
 
+# Create pkg-config file
+include(build/JoinPaths.cmake)
+# from: https://github.com/jtojnar/cmake-snips#concatenating-paths-when-building-pkg-config-files
+join_paths(UVATLAS_INCLUDEDIR_FOR_PKG_CONFIG "\${prefix}" "${CMAKE_INSTALL_INCLUDEDIR}")
+join_paths(UVATLAS_LIBDIR_FOR_PKG_CONFIG "\${prefix}"     "${CMAKE_INSTALL_LIBDIR}")
+
+set(UVATLAS_DEP_L "")
```
Comment (walbourn) [inline CMakeLists.txt:246]: I believe this should be:

```
set(UVATLAS_DEP_L "")
if(ENABLE_USE_EIGEN)
  list(APPEND UVATLAS_DEP_L "eigen3")
endif()
if(directxmath_FOUND)
  list(APPEND UVATLAS_DEP_L "DirectXMath")
endif()
if(directx-headers_FOUND)
  list(APPEND UVATLAS_DEP_L "DirectX-Headers")
endif()
list(LENGTH UVATLAS_DEP_L DEP_L)
if(DEP_L)
  string(REPLACE ";" ", " UVATLAS_DEP " ${UVATLAS_DEP_L}")
endif()
```

Comment (JPeterMugaas) [reply CMakeLists.txt:246]: Fixed but requires https://github.com/microsoft/DirectXMath/pull/174 .

**Review by walbourn — APPROVED** [review]
Thank you very much for your contribution.

**Pull request merged**